### DISCUSSION TITLE
Loading plugins dynamically

### DISCUSSION THREAD

**Discussion by Trobeloo** [quartznet/quartznet#2267] [Q&A]
Hello,

I need to load/use plugin dynamically (without referncing the aseembly) by just declaring them in the config file. By defaul it doesn't seems to work (file not found error).
I see that I can implement a loadHelper class to resolve that, but LoadHelper is not used when setting SchedulerPlugins:
https://github.com/quartznet/quartznet/blob/7d525f21a3166c3b595508d345e23313fb9e8d4d/src/Quartz/Impl/StdSchedulerFactory.cs#L797

Is there any particular reason for that, can it be changed to the below code? 
var pluginTypeType = loadHelper.LoadType(plugInType) ?? throw new ConfigurationException($"Could not load plugin type {plugInType}");

Jacques

**Comment by Trobeloo** [comment]
I am using .Net 8.
There are some nice blog posts about GetType in .Net core: 
https://jeremybytes.blogspot.com/2020/01/using-typegettype-with-net-core.html
https://jeremybytes.blogspot.com/2020/01/typegettype-functionality-has-not.html

**Comment by Trobeloo** [comment]
Those blog posts reference a Microsoft article about "Create a .NET Core application with plugins"
https://learn.microsoft.com/en-us/dotnet/core/tutorials/creating-app-with-plugin-support

**Comment by lahma** [comment]
Can you show how you configure and get FileNotFound, is it Quartz config file or something you've built on top?

**Comment by Trobeloo** [comment]
I am using the source code of Quartz.Server 
The plugin is adapted from https://www.codeproject.com/Articles/1017396/Running-jobs-sequentially-using-Quartz-Net-with-th
Quartz.config is modified to add the following line:
quartz.plugin.PluginExample.type = SequentialQuartz_POC.PluginExample, SequentialQuartz_POC
The assemblyr "SequentialQuartz_POC.dll" is copied in the compile directory of Quartz.Server

Error is:
Quartz.SchedulerException: 'SchedulerPlugin of type 'SequentialQuartz_POC.PluginExample, SequentialQuartz_POC' could not be instantiated.'
Inner Exception
FileNotFoundException: Could not load file or assembly 'SequentialQuartz_POC, Culture=neutral, PublicKeyToken=null'. The system cannot find the file specified.

If I add a refrence to the assembly it works.

If I do not add a reference but add those lines to Main the it works
 string assmblyPath = AppDomain.CurrentDomain.BaseDirectory + "SequentialQuartz_POC.dll";
 AssemblyLoadContext.Default.LoadFromAssemblyPath(assmblyPath);
 Type pluginType = Type.GetType("SequentialQuartz_POC.PluginExample, SequentialQuartz_POC");

**Comment by lahma** [comment]
#2268 hopefully improves the situation
